File: coding-practice/50 Coding Challenges part 1/C#/problem21.cs
// Coding challenge #21 : Rotate an array to the right 1 position
using System;
class problem21
{
    static int[] rotate_right(int[] input){
        int[] output = new int[input.Length];
        for(int i=0; i<input.Length-1; i++){
            output[i] = input[i+1];
        }
        output[input.Length-1] = input[0];
        return output;
    }

    static void Main(string[] args){
        int[] array = {1, 2, 3, 4, 5};
        int[] output = new int[array.Length];
        output = rotate_right(array);
        Console.Write("The rotated version of ["+String.Join(", ",array)+"] is : ["+String.Join(", ", output)+"]");
    }
}

File: coding-practice/50 Coding Challenges part 1/Python/problem21.py
# Coding challenge 21 : Rotate an array to the right 1 position

# The initial array
array = [1, 2, 3, 4, 5]

# Create a new empty array
newArray = []

# Set the last value of the array to be the first value of the newArray
newArray.append(array[len(array) - 1])

# This looping is inserting the rest of the array's value to the newArray
for i in range(0, 4):
    newArray.append(array[i])

# print the newArray
print(newArray)